OpenRA.Mods.Common/Traits/RepairableNear.cs

```diff
@@ -84,5 +85,17 @@ public void ResolveOrder(Actor self, Order order)
 				self.SetTargetLine(target, Color.Green, false);
 			}
 		}
+
+		public Actor FindRepairBuilding(Actor self)
+		{
+			var repairBuilding = self.World.ActorsWithTrait<RepairsUnits>()
+				.Where(a => !a.Actor.IsDead && a.Actor.IsInWorld
+					&& a.Actor.Owner.IsAlliedWith(self.Owner) &&
+					info.Buildings.Contains(a.Actor.Info.Name))
+				.OrderBy(p => (self.Location - p.Actor.Location).LengthSquared);
+
+			// Worst case FirstOrDefault() will return a TraitPair<null, null>, which is OK.
+			return repairBuilding.FirstOrDefault().Actor;
```

Comment:
I didn't check this, it is adapted copy from Repairable.FindRepairBuilding(). It contains this comment. Should I check it or can we assume it is correct?